test/abstract_client.ts

```diff
@@ -2101,64 +2126,79 @@ export default function abstractTest(server, config, ports) {
 		})
 
 		const reschedulePing = (reschedulePings: boolean) => {
-			it(`should ${
-				!reschedulePings ? 'not ' : ''
-			}reschedule pings if publishing at a higher rate than keepalive and reschedulePings===${reschedulePings}`, function _test(t, done) {
-				const intervalMs = 3000
-				const client = connect({
-					keepalive: intervalMs / 1000,
-					reschedulePings,
-				})
-
-				const spyReschedule = sinon.spy(
-					client,
-					'_reschedulePing' as any,
-				)
+			it(
+				`should ${
+					!reschedulePings ? 'not ' : ''
+				}reschedule pings if publishing at a higher rate than keepalive and reschedulePings===${reschedulePings}`,
+				{
+					timeout: 4000,
+				},
+				function _test(t, done) {
+					clock.restore()
 
-				let received = 0
+					teardownHelper.add(
+						{
+							executeOnce: true,
+							order: 1,
+						},
+						() => {
+							if (locaClock) {
+								locaClock.restore()
+							}
+						},
+					)
 
-				client.on('packetreceive', (packet) => {
-					if (packet.cmd === 'puback') {
-						process.nextTick(() => {
-							clock.tick(intervalMs)
+					const locaClock = sinon.useFakeTimers({
```

Comment:
Just a small typo here, fix it then we can merge :) 

```suggestion
					const localClock = sinon.useFakeTimers({
```

Reply:
Done.